posts/2021/10/numpy_benchmarking_blog.md

```diff
@@ -0,0 +1,165 @@
+.. slug: numpy-benchmarking
+.. date: 2021-09-27 12:23:40 UTC-05:00
+.. author: Khushi Agrawal
+.. tags: NumPy, Pythran, Numba, Transonic
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+.. previewimage:
+-->
+
+In this blog post, I'll be talking about my journey in Quansight. 
+I want to share all things I was involved in and accomplished. 
+What issues I faced, and most importantly, what were awesome life hacks I learned during this period.
+
+First of all, I'd like to express my gratitude to the whole team 
+for allowing me to be a part of such a great team. 
+My work was majorly focused on providing performance benchmarks to NumPy in realistic situations. 
+The target was to show the world that NumPy is efficient in handling quasi real-life situations too.
+
+<p align="center">
+      <img src = "/images/2021/10/journey.jpeg" alt = "A word cloud with themes, open-source projects and people mentioned throughout the blog post. Each is stylized using a different font, most of them calligraphical.">
+</p>
+
+<!-- TEASER_END -->
+
+## My Experience
+My work was broadly divided majorly into the following categories:
+
+- **N-Body Problem**: The [N-Body problem](https://en.wikipedia.org/wiki/N-body_problem) is one of the most famous 
+and universally accepted problems for benchmarking. 
+I was given this as a problem statement to work on the project. 
+I started my work with a theoretical understanding of the problem. 
+I took reference from [Pierre Augier] remarkable work on 
+[n-body](https://github.com/paugier/nbabel) problem. I'd love to thank him.
+It was a fun learning part for me to connect the scientific part with the programming world. 
+I implemented the N-Body problem in Python, C, and C++.
+
+- **Compiled Methods**: This part was the most exciting part of the project. 
+I got introduced to various compiled methods like [Cython](https://cython.readthedocs.io/en/latest/), 
+[Pythran](https://pythran.readthedocs.io/en/latest/), [Numba](http://numba.pydata.org/), [Transonic](https://transonic.readthedocs.io/en/latest/). 
+It was the first time when I got to know about accelerators. 
+I visited their documentations, GitHub, and searched popular blogs about compiled methods. 
+I loved playing with them. I read the theory part, looked into the examples, 
+and implemented them in my editor. It was a great learning experience for me 
+to get familiar with compiled methods. 
+I implemented the jitted compilation mode in Pythran and Numba for benchmarking via Transonic's support.
+
+- **Visualization**: I used the [Matplotlib](https://matplotlib.org/) library for visualizing the benchmarking results. 
+I tried various plots to verify which one suits best, 
+like scatter plots, box plots, line charts, the combination of the scatter plots and line charts, etc. 
+But these were not good to go. 
+Those plots either lacked clarity or were not capable of providing significant results. 
+We finalized decided on horizontally stacked bar charts.
```

Comment:
I thought we are using a [vertical grouped bar chart](https://matplotlib.org/stable/gallery/index.html)
```suggestion
We finalized decided on two bar charts, with different vertical scales to accomodate the vastly different performance of Python vs. the compiled methods. We also normalized the data to show trends as the number of particles increased.
```